src/main/scala/lms/thirdparty/mpi.scala

```diff
@@ -37,13 +37,18 @@ trait MPIOps extends CMacro with LibStruct with LibFunction { b: Base =>
   }
 
   // this is how we bind to library structs with field read access
-  abstract class DataStructure1
+  class DataStructure1
+  // 1. define an implicit value for c-type registration (internally it is a CustomManifest)
+  implicit val registCTypeForDataStructure1 = registerCType[DataStructure1]("DataStructure1")
+  // 2. define the instantiation function
+  def dataStructure1: Rep[DataStructure1] = newStruct[DataStructure1]
+  // 3. add any field read access functions
   implicit class DataStructure1Ops(x: Rep[DataStructure1]) {
     val fieldA: Rep[Int] = readField[DataStructure1, Int](x, "fieldA")
```

Comment:
is it should be a `def`?

Reply:
better to be a `def` I think :) 

using `val` is assuming that no more changes happen to the struct, which is sort of true here (and helps with CSE), but quite constraint.

Reply:
fixed